packages/snaps-utils/src/checksum.ts

```diff
@@ -9,8 +9,20 @@ import { VirtualFile } from './virtual-file/VirtualFile';
  * @param bytes - The byte array to calculate the checksum for.
  * @returns A single sha-256 checksum.
  */
-export function checksum(bytes: VirtualFile | Uint8Array | string): Uint8Array {
-  const value = bytes instanceof VirtualFile ? bytes.value : bytes;
+export async function checksum(
+  bytes: VirtualFile | Uint8Array | string,
+): Promise<Uint8Array> {
+  const unwrapped = bytes instanceof VirtualFile ? bytes.value : bytes;
+  const value =
+    typeof unwrapped === 'string' ? stringToBytes(unwrapped) : unwrapped;
```

Comment:
It seems cleaner to extract this into a separate function and get rid of the ternaries.